lib/parser.mly

```diff
@@ -1,599 +0,0 @@
-
-parameters_block:
-  | PARAMETERSBLOCK LBRACE tvd=list(top_var_decl_no_assign) RBRACE
-    { grammar_logger "parameters_block" ; tvd }
-
-transformed_parameters_block:
-  | TRANSFORMEDPARAMETERSBLOCK LBRACE tvds=list(top_vardecl_or_statement) RBRACE
-    { grammar_logger "transformed_parameters_block" ; tvds }
-
-model_block:
-  | MODELBLOCK LBRACE vds=list(vardecl_or_statement) RBRACE
-    { grammar_logger "model_block" ; vds  }
-
-generated_quantities_block:
-  | GENERATEDQUANTITIESBLOCK LBRACE tvds=list(top_vardecl_or_statement) RBRACE
-    { grammar_logger "generated_quantities_block" ; tvds }
-
-(* function definitions *)
-identifier:
-  | id=IDENTIFIER
-    {
-      grammar_logger ("identifier " ^ id) ;
-      {name=id; id_loc=Location ($startpos, $endpos)}
-    }
-  | TRUNCATE
-    {
-      grammar_logger "identifier T" ;
-      {name="T"; id_loc=Location ($startpos, $endpos)}
-    }
-
-function_def:
-  | rt=return_type name=identifier LPAREN args=separated_list(COMMA, arg_decl)
-    RPAREN b=statement
-    {
-      grammar_logger "function_def" ;
-      UntypedStmt (FunDef {returntype = rt; funname = name;
-                           arguments = args; body=b;},
-      initialize_stmt_meta $startpos $endpos)
-    }
-
-return_type:
-  | VOID
-    { grammar_logger "return_type VOID" ; Void }
-  | ut=unsized_type
-    {  grammar_logger "return_type unsized_type" ; ReturnType ut }
-
-arg_decl:
-  | od=option(DATABLOCK) ut=unsized_type id=identifier
-    {  grammar_logger "arg_decl" ;
-       match od with None -> (TParam, ut, id) | _ -> (TData, ut, id)  }
-
-unsized_type:
-  | bt=basic_type ud=option(unsized_dims)
-    {  grammar_logger "unsized_type" ;
-       let reparray n x =
-         let rec repeat n f x =
-           if n <= Int64.zero then x else repeat (Int64.pred n) f (f x) in
-         repeat n (fun y -> Array y) x in
-       let size =
-         match ud with Some d -> Int64.succ (Int64.of_int d) | _ -> Int64.zero
-       in
-       reparray size bt    }
-
-basic_type:
-  | INT
-    {  grammar_logger "basic_type INT" ; Int  }
-  | REAL
-    {  grammar_logger "basic_type REAL"  ; Real }
-  | VECTOR
-    {  grammar_logger "basic_type VECTOR" ; Vector  }
-  | ROWVECTOR
-    {  grammar_logger "basic_type ROWVECTOR" ; RowVector  }
-  | MATRIX
-    {  grammar_logger "basic_type MATRIX" ; Matrix  }
-
-unsized_dims:
-  | LBRACK cs=list(COMMA) RBRACK
-    { grammar_logger "unsized_dims" ; List.length(cs) }
-
-(* declarations *)
-var_decl:
-  | sbt=sized_basic_type id=identifier d=option(dims)
-    ae=option(pair(ASSIGN, expression)) SEMICOLON
-    { grammar_logger "var_decl" ;
-      let sizes = match d with None -> [] | Some l -> l in
-      match ae with
-      | Some a ->
-          UntypedStmt
-            ( VDeclAss
-                {sizedtype= reducearray (sbt, sizes); identifier= id; value= snd a}
-            , initialize_stmt_meta $startpos $endpos )
-      | _ ->
```

Comment:
I think might be better to write `None ->` just to be explicit

Reply:
Agreed. Done.